File: javascript/time.js
/**
 *  @param {function} funcName Function to be executed
 */
function checkFuncTime(funcName) {
  const start_time = new Date().getMilliseconds();
  try {
    funcName();
  } catch (e) {
    console.log(e);
  }
  const end_time = new Date().getMilliseconds();
  // compute the func time
  console.log(
    "Function " +
      funcName.name +
      " took " +
      (end_time - start_time) +
      " milliseconds to execute."
  );
}

/**
 * function
 */
function longExecutingFunc() {
    var array = [], total = 10000000, i=0;
    while(i<=total) {
        array.push(Math.random())
        i+=1
    }
    console.log("I am long fun!")
}

/**
 * small running function
 */
function shortRunningFunc() {
  console.log("I am short func!");
}

checkFuncTime(shortRunningFunc);

checkFuncTime(longExecutingFunc);

File: python/time.py
import time
from functools import wraps


def func_time(func):
    """
    Compute function running time
    """
    @wraps(func)
    def checkTime(*args, **kwargs):
        start_time = time.process_time()
        func(*args, **kwargs)
        end_time = time.process_time()
        print('It took ', (end_time - start_time),
              ' milliseconds to run ', func.__name__)
    return checkTime


@func_time
def long_executing_func():
    print('I am long func!')
    arr = [i for i in range(10000000)]


@func_time
def short_running_func():
    print('I am short func!')
    arr = [i for i in range(10000)]


short_running_func()
long_executing_func()
